File: lib/java/opentoken/src/test/java/com/truveta/opentoken/attributes/AttributeLoaderTest.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.attributes;

import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

import com.truveta.opentoken.attributes.general.RecordIdAttribute;
import com.truveta.opentoken.attributes.person.LastNameAttribute;

class AttributeLoaderTest {

    @Test
    void loadAttributes_ShouldLoadAttributes() {
        var attributesSet = AttributeLoader.load();

        var recordIdAttribute = attributesSet.stream()
                .filter(RecordIdAttribute.class::isInstance)
                .findFirst();
        assertTrue(recordIdAttribute.isPresent(), "RecordIdAttribute should be loaded");

        var lastNameAttribute = attributesSet.stream()
                .filter(LastNameAttribute.class::isInstance)
                .findFirst();
        assertTrue(lastNameAttribute.isPresent(), "LastNameAttribute should be loaded");
    }
}

File: lib/python/opentoken/src/test/opentoken/attributes/attribute_loader_test.py
"""
Copyright (c) Truveta. All rights reserved.
"""

import pytest
from opentoken.attributes.attribute_loader import AttributeLoader
from opentoken.attributes.general.record_id_attribute import RecordIdAttribute
from opentoken.attributes.person.last_name_attribute import LastNameAttribute


class TestAttributeLoader:
    """Test cases for AttributeLoader."""

    def test_load_attributes_should_load_attributes(self):
        """Test that load_attributes loads the expected attributes."""
        attributes_set = AttributeLoader.load()

        # Find RecordIdAttribute in the loaded attributes
        record_id_attribute = next(
            (attr for attr in attributes_set if isinstance(attr, RecordIdAttribute)),
            None
        )
        assert record_id_attribute is not None, "RecordIdAttribute should be loaded"

        # Find LastNameAttribute in the loaded attributes
        last_name_attribute = next(
            (attr for attr in attributes_set if isinstance(attr, LastNameAttribute)),
            None
        )
        assert last_name_attribute is not None, "LastNameAttribute should be loaded"